htdocs/mri/jiv/get_file.php

```diff
@@ -185,9 +161,67 @@
             pathinfo($DownloadFilename, PATHINFO_EXTENSION);
 
     }
-    header("Content-Disposition: attachment; filename=$DownloadFilename");
 }
+header("Content-Disposition: attachment; filename=$DownloadFilename");
```

Comment:
It's changing the content disposition, not just the filename.

Reply:
And I think we just need someone 👀with enough Imaging Browser knowledge to know whether this is actually a big deal or not.

Reply:
@Jkat Do you think it would make sense for me to bring it up in a LORIS meeting?

Reply:
Previously, for cases `png`, `jpg`, `header`, and `raw_byte.gz`, the variable `$DownloadFilename` never gets set, and therefore this block https://github.com/aces/Loris/pull/4096/files#diff-a81339fada2f49a25145b8abb4a4e6f6R148 wouldn't get executed.
But now you've explicitly set `$DownloadFilename = basename($File);` two lines above, so therefore in theory that block would always get executed. Then later `header("Content-Disposition: attachment; filename=$DownloadFilename");` would get triggered for all cases regardless whether they're `png`, `jpg`, `header`, and `raw_byte.gz`.

Reply:
I think we can remove the `if (!empty($DownloadFilename)) {` since it will indeed always be true.

I am not convinced that the change of logic would break anything. BTW, header and raw_byte.gz corresponds to the JIV format which have been removed from the imaging pipeline and LORIS a few release back so I am not worries about those 2 file types. As for png and jpg, I am not sure for what they are used for but I don't think the changes here are a major issue (although I may be wrong).

Hope this helps!

Reply:
As @jkat says, in the `png`, `jpg`, `header`, and `raw_bytes.gz` cases the variable never got set and the `Content-Disposition` header was therefore not sent. Regardless of if it breaks anything, it's semantically wrong and a regression. It doesn't make sense to remove the check because it doesn't make sense for types that are displayed inline to have an attachment content disposition. png and jpgs, at the very least, are not there to be downloaded as an attachment, they're to be displayed as part of the page. (I believe that's also true of the other 2 files which are for jiv, but I'm less certain about them.)

Reply:
I get it now. I figured that checking `empty()` on a potentially unset variable was a mistake. I didn't piece it together that the attachment Content-Disposition was being triggered that way.

I reverted the change and also made sure that `$DownloadFilename` starts as an empty string. Also added some comments to make this more clear.